google/resource_container_node_pool.go

```diff
@@ -155,11 +151,57 @@ var schemaNodePool = map[string]*schema.Schema{
 	},
 }
 
-func resourceContainerNodePoolCreate(d *schema.ResourceData, meta interface{}) error {
-	containerApiVersion := getContainerApiVersion(d, ContainerNodePoolBaseApiVersion, ContainerNodePoolVersionedFeatures)
-	config := meta.(*Config)
+type NodePoolInformation struct {
+	project  string
+	location string
+	cluster  string
+}
+
+func (nodePoolInformation *NodePoolInformation) fullyQualifiedName(nodeName string) string {
+	return fmt.Sprintf(
+		"projects/%s/locations/%s/clusters/%s/nodePools/%s",
+		nodePoolInformation.project,
+		nodePoolInformation.location,
+		nodePoolInformation.cluster,
+		nodeName,
+	)
+}
 
+func (nodePoolInformation *NodePoolInformation) parent() string {
+	return fmt.Sprintf(
+		"projects/%s/locations/%s/clusters/%s",
+		nodePoolInformation.project,
+		nodePoolInformation.location,
+		nodePoolInformation.cluster,
+	)
+}
+
+func (nodePoolInformation *NodePoolInformation) lockKey() string {
+	return containerClusterMutexKey(nodePoolInformation.project,
+		nodePoolInformation.location, nodePoolInformation.cluster)
+}
+
+func extractNodePoolInformation(d *schema.ResourceData, config *Config) (*NodePoolInformation, error) {
 	project, err := getProject(d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	location, err := getLocation(d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &NodePoolInformation{
+		project:  project,
+		location: location,
+		cluster:  d.Get("cluster").(string),
```

Comment:
This is giving panics:

Edited for space:
```
panic: interface conversion: interface {} is nil, not string

github.com/terraform-providers/terraform-provider-google/google.extractNodePoolInformation(0xc42030ae00, 0xc420658000, 0x0, 0xc42010d160, 0x0)
resource_container_node_pool.go:198 +0x262
github.com/terraform-providers/terraform-provider-google/google.nodePoolUpdate(0xc42030ae00, 0x18126a0, 0xc420658000, 0xc4202ee600, 0x17, 0xc420530970, 0xc, 0xa, 0xc42041b680, 0x1a)
resource_container_node_pool.go:507 +0x68
github.com/terraform-providers/terraform-provider-google/google.resourceContainerClusterUpdate(0xc42030ae00, 0x18126a0, 0xc420658000, 0x0, 0xc4207dce08)
resource_container_cluster.go:1136 +0x74c
```

Reply:
Is that from running the test suite or related to the autoscaling test I mentioned?

Reply:
From running the test suite.  Same panic in 

`TestAccContainerCluster_withAdditionalZones`
`TestAccContainerCluster_withAddons`
`TestAccContainerCluster_withLegacyAbac`
`TestAccContainerCluster_withMasterAuthorizedNetworksConfig`

Possibly others - tests aren't done yet.

I'm looking at that test you mentioned as well.

Reply:
OK, I can take a look at that. I wonder if it's because the `cluster` key requires a prefix to resolve in the map?

Reply:
I'm betting that's it.

Reply:
I've passed along the prefix to it now and I'm running `TestAccContainerCluster_withAddons` so I'll see what the output is.

If you can provide any insight into the autoscaling test above that will be awesome 👍 

I'll try my best to do some work this evening for the good of regionals!

Reply:
It didn't work adding the prefix, attaching the debugger to see if I can see other issues.

Reply:
So it is the cluster key. That key does not exist from the perspective from the perspective of a cluster, I'm looking at at fix now

Reply:
@ndmckinley I have pushed a fix tested it with `TestAccContainerCluster_withAddons` and it's passing.

See commit for info.